tests/test_interpolation.py

```diff
@@ -257,36 +276,92 @@ def test_env_interpolation(
         # yaml strings are not getting parsed by the env resolver
         ("foo: bar", "foo: bar"),
         ("foo: \n - bar\n - baz", "foo: \n - bar\n - baz"),
+        # more advanced uses of the grammar
+        ("ab \\{foo} cd", "ab \\{foo} cd"),
+        ("ab \\\\{foo} cd", "ab \\\\{foo} cd"),
+        ("'\\${other_key}'", "${other_key}"),  # escaped interpolation
+        ("'ab \\${other_key} cd'", "ab ${other_key} cd"),  # escaped interpolation
+        ("[1, 2, 3]", [1, 2, 3]),
+        ("{a: 0, b: 1}", {"a": 0, "b": 1}),
+        ("  123  ", "  123  "),
+        ("  1 2 3  ", "  1 2 3  "),
+        ("\t[1, 2, 3]\t", "\t[1, 2, 3]\t"),
+        ("[\t1, 2, 3\t]", [1, 2, 3]),
+        ("   {a: b}\t  ", "   {a: b}\t  "),
+        ("{   a: b\t  }", {"a": "b"}),
+        ("'123'", "123"),
+        ("${env:my_key_2}", 456),  # can call another resolver
     ],
 )
 def test_env_values_are_typed(value: Any, expected: Any) -> None:
     try:
         os.environ["my_key"] = value
+        os.environ["my_key_2"] = "456"
         c = OmegaConf.create(dict(my_key="${env:my_key}"))
         assert c.my_key == expected
     finally:
         del os.environ["my_key"]
+        del os.environ["my_key_2"]
+
+
+def test_env_node_interpolation() -> None:
+    # Test that node interpolations are not supported in env variables.
+    try:
+        os.environ["my_key"] = "${other_key}"
+        c = OmegaConf.create(dict(my_key="${env:my_key}", other_key=123))
+        with pytest.raises(ConfigAttributeError):
+            c.my_key
+    finally:
+        del os.environ["my_key"]
```

Comment:
mokneypatch.

Reply:
Oh right, I learned about that along the way and forgot to go back to older tests... done in 540e335